tests/test_main.py

```
@@ -607,6 +607,203 @@ def _capture_create_task(coro: Any, *args: Any, **kwargs: Any) -> Any:
+        mock_connect_matrix.return_value = mock_matrix_client
+        mock_connect_meshtastic.return_value = MagicMock()
+
+        async def _release_startup_drain() -> None:
+            await asyncio.sleep(0.01)
+            startup_drain_complete_event.set()
+
+        def _write_ready_side_effect() -> None:
+            readiness_calls.append("ready")
+            shutdown_event.set()
+
+        def _capture_main_info(message: str, *args: Any, **_kwargs: Any) -> None:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)